### ISSUE TITLE
Point positioning changes depending on left or right side

### ISSUE THREAD

**Issue by andrewheiss** [erocoar/gghalves#5]
For some reason, `geom_half_point` places points all the way up to the center when positioned on the left, but not when positioned on the right:

``` r
library(ggplot2)
library(gghalves)

# Points go all the way up to the center
ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "r") + 
  geom_half_point(side = "l")
```

![](https://i.imgur.com/Bq7xalU.png)

``` r

# Points don't go the center!
ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "l") + 
  geom_half_point(side = "r")
```

![](https://i.imgur.com/Yrqq1Hc.png)

**Comment by andrewheiss** [comment]
I tried fiddling with the `width` argument to `PositionJitter`, but it doesn't seem to have any effect, which could possibly be another bug:

``` r
library(ggplot2)
library(gghalves)

# Maybe force a bigger width?

# The width argument seems to have no effect on anything. These are all the same:
ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "l") + 
  geom_half_point(side = "r", transformation = PositionJitter,
                  transformation_params = list(width = 1, seed = 1234))
```

![](https://i.imgur.com/5IYeZ5u.png)

``` r

ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "l") + 
  geom_half_point(side = "r", transformation = PositionJitter,
                  transformation_params = list(width = 0.0001, seed = 1234))
```

![](https://i.imgur.com/3YQxoSr.png)

``` r

ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "l") + 
  geom_half_point(side = "r", transformation = PositionJitter,
                  transformation_params = list(width = 10, seed = 1234))
```

![](https://i.imgur.com/S3Zllsi.png)

**Comment by andrewheiss** [comment]
(This is using the development version of ggplot2 (3.3.0.9000) and the most current gghalves package from GitHub)

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       macOS Catalina 10.15.3      
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/New_York            
#>  date     2020-02-13                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version    date       lib
#>  assertthat    0.2.1      2019-03-21 [1]
#>  backports     1.1.5      2019-10-02 [1]
#>  callr         3.4.2      2020-02-12 [1]
#>  cli           2.0.1      2020-01-08 [1]
#>  colorspace    1.4-1      2019-03-18 [1]
#>  crayon        1.3.4      2017-09-16 [1]
#>  curl          4.3        2019-12-02 [1]
#>  desc          1.2.0      2018-05-01 [1]
#>  devtools      2.0.2      2019-04-08 [1]
#>  digest        0.6.24     2020-02-12 [1]
#>  dplyr         0.8.4      2020-01-31 [1]
#>  evaluate      0.14       2019-05-28 [1]
#>  fansi         0.4.1      2020-01-08 [1]
#>  farver        2.0.3      2020-01-16 [1]
#>  fs            1.3.1      2019-05-06 [1]
#>  gghalves    * 0.0.1.9000 2020-02-13 [1]
#>  ggplot2     * 3.3.0.9000 2020-02-13 [1]
#>  glue          1.3.1      2019-03-12 [1]
#>  gtable        0.3.0      2019-03-25 [1]
#>  highr         0.8        2019-03-20 [1]
#>  htmltools     0.4.0      2019-10-04 [1]
#>  httr          1.4.1      2019-08-05 [1]
#>  knitr         1.23       2019-05-18 [1]
#>  labeling      0.3        2014-08-23 [1]
#>  lifecycle     0.1.0      2019-08-01 [1]
#>  magrittr      1.5        2014-11-22 [1]
#>  memoise       1.1.0      2017-04-21 [1]
#>  mime          0.9        2020-02-04 [1]
#>  munsell       0.5.0      2018-06-12 [1]
#>  pillar        1.4.3      2019-12-20 [1]
#>  pkgbuild      1.0.6      2019-10-09 [1]
#>  pkgconfig     2.0.3      2019-09-22 [1]
#>  pkgload       1.0.2      2018-10-29 [1]
#>  prettyunits   1.1.1      2020-01-24 [1]
#>  processx      3.4.2.9000 2020-02-12 [1]
#>  ps            1.3.2      2020-02-13 [1]
#>  purrr         0.3.2      2019-03-15 [1]
#>  R6            2.4.1      2019-11-12 [1]
#>  Rcpp          1.0.3      2019-11-08 [1]
#>  remotes       2.0.4      2019-04-10 [1]
#>  rlang         0.4.4      2020-01-28 [1]
#>  rmarkdown     1.13       2019-05-22 [1]
#>  rprojroot     1.3-2      2018-01-03 [1]
#>  scales        1.1.0      2019-11-18 [1]
#>  sessioninfo   1.1.1      2018-11-05 [1]
#>  stringi       1.4.5      2020-01-11 [1]
#>  stringr       1.4.0      2019-02-10 [1]
#>  testthat      2.3.1      2019-12-01 [1]
#>  tibble        2.1.3      2019-06-06 [1]
#>  tidyselect    0.2.5      2018-10-11 [1]
#>  usethis       1.5.0.9000 2019-06-22 [1]
#>  withr         2.1.2      2018-03-15 [1]
#>  xfun          0.8        2019-06-25 [1]
#>  xml2          1.2.0.9000 2019-06-18 [1]
#>  yaml          2.2.1      2020-02-01 [1]
#>  source                               
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  Github (erocoar/gghalves@22c0780)    
#>  Github (tidyverse/ggplot2@b434351)   
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  Github (r-pkgs/processx@4e3715a)     
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  Github (r-lib/usethis@28cd06e)       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  Github (HenrikBengtsson/xml2@618bc74)
#>  CRAN (R 3.6.0)                       
#> 
#> [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

</details>

**Comment by andrewheiss** [comment]
Ack, sorry :)

Finally, `width` doesn't seem to do anything if the points are on the left either:

``` r
library(ggplot2)
library(gghalves)

# Width also doesn't do anything when points are on the left
ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "r") + 
  geom_half_point(side = "l", transformation = PositionJitter,
                  transformation_params = list(width = 1, seed = 1234))
```

![](https://i.imgur.com/cnNcmEx.png)

``` r

ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "r") + 
  geom_half_point(side = "l", transformation = PositionJitter,
                  transformation_params = list(width = 0.0001, seed = 1234))
```

![](https://i.imgur.com/iDpgqlY.png)

``` r

ggplot(iris, aes(x = Species, y = Sepal.Length)) +
  geom_half_boxplot(side = "r") + 
  geom_half_point(side = "l", transformation = PositionJitter,
                  transformation_params = list(width = 10, seed = 1234))
```

![](https://i.imgur.com/XGSjQJj.png)

**Comment by erocoar** [comment]
Hi, thanks for your feedback! This happened because the points were (incorrectly) scaled _after_ they had already been jittered. Actually, I don't think this is even necessary so I have dropped it now and `width` should work again. 

Also, I introduced a new parameter `range_scale`. When `width` is not specified (`NULL`), it determines the spread of the jitter. Default is `0.5`, i.e. the jitter uses half of the allotted x-axis width. `1` would use all the width, etc. 

```r
library(cowplot)
library(gghalves)
ranges <- c(1, .5, .25)
plts <- lapply(ranges, function(x) {
  ggplot(iris, aes(x = Species, y = Sepal.Length)) + 
    geom_half_boxplot(side = "l") + 
    geom_half_point(side = "r", range_scale = x)
})

cowplot::plot_grid(plotlist = plts, ncol = 3, labels = ranges)
```

![range_scale_plt](https://user-images.githubusercontent.com/16050768/74527646-8fe93b80-4f26-11ea-917b-fca7472a5a2e.png)

I'll update it on CRAN soon
